src/Modal.js

```diff
@@ -162,9 +165,34 @@ class Modal extends React.Component {
       }
     }
 
+    document.removeEventListener('focus', this.trapFocus);
     this._isMounted = false;
   }
 
+  trapFocus (ev) {
+    if (!this._element)
+        return ;
+
+    if (this._dialog && this._dialog.parentNode === ev.target)
+        return ;
+
+    if (this.modalCount < Modal.openCount)
+        return ;
+
+    const children = this.getFocusableChildren();
+
+    for (let i = 0; i < children.length; i++) {
+        if (children[i] === ev.target)
+            return ;
+    }
+
+    if (children.length > 0) {
+      ev.preventDefault();
+      ev.stopPropagation();
+      children[0].focus();
+    }
+  }
+
```

Comment:
I don't believe we need this `trapFocus` function now that you're only handling keyboard on topmost modal?

ie. I think that your below fix of the early return in `handleTab` should be enough to fix the issue?

Reply:
The trapFocus function was implemented considering #1679. This issue reports that the user can set focus to elements outside of the Modal after focusing the address bar. I tried addressing both issues in the same commit, sorry for that lack of organization.

Reply:
Ah ok makes sense now.  Can you fix the formatting here?  Indents are a bit off.   Also add some comments next to each branch inside the `trapFocus` function to let us know how it works and what each branch is preventing.